Airwallex/Payment/Sources/CollectionViewSectionManager.swift

```diff
@@ -0,0 +1,142 @@
+                    return nil
+                }
+                return controller.layout(environment: environment)
+            },
+            configuration: listConfiguration
+        )
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.delegate = self
+        
+        if let boundaryItems {
+            for item in boundaryItems {
+                collectionView.register(
+                    item.reusableView,
+                    forSupplementaryViewOfKind: item.elementKind,
+                    withReuseIdentifier: item.reusableView.reuseIdentifier
+                )
+            }
+        }
+        
+       
+        diffableDataSource = UICollectionViewDiffableDataSource<SectionType, ItemType>(
+            collectionView: collectionView,
+            cellProvider: { [weak self] collectionView, indexPath, itemIdentifier in
+                guard let self, let sectionIndex = self.sections[safe: indexPath.section],
+                      let sectionController = self.sectionControllers[sectionIndex] else {
+                    assert(false, "invalid section index")
+                    return UICollectionViewCell()
+                }
+                return sectionController.cell(for: collectionView, item: itemIdentifier, at: indexPath)
+            }
+        )
+        
+        diffableDataSource.supplementaryViewProvider = {[weak self] (collectionView, elementKind, indexPath) in
+            if elementKind == "collection-header-element-kind" {
+                return collectionView.dequeueReusableSupplementaryView(ofKind: elementKind, withReuseIdentifier: LabelHeader.reuseIdentifier, for: indexPath)
+            }
+            guard let self,
+                  let section = self.sections[safe: indexPath.section],
+                  let sectionController = self.sectionControllers[section] else {
+                assert(false, "UI not consistance with data source")
+                return UICollectionReusableView()
+            }
+            
+            return sectionController.supplementaryView(for: collectionView, ofKind: elementKind, at: indexPath)
+        }
+        self.context = CollectionViewContext(viewController: viewController, dataSource: diffableDataSource)
+    }
+    
+    func reloadData() {
+        guard let sectionDataSource else { return }
+        sections = sectionDataSource.sections()
+        for section in sections {
+            if sectionControllers[section] != nil { continue }
+            var sectionController = sectionDataSource.sectionController(for: section)
+            sectionController.bind(context: context)
+            sectionController.registerReusableViews(to: collectionView)
+            sectionControllers[section] = sectionController
+            
+        }
+        var snapshot = NSDiffableDataSourceSnapshot<SectionType, ItemType>()
+        snapshot.appendSections(sections)
+        for section in sections {
```

Comment:
I understand we're loading all the data to the snapshot, but do we actually need two loops through sections?

Reply:
addressed